mcweb/backend/sources/tasks.py

```diff
@@ -315,3 +319,89 @@ def schedule_scrape_source(source_id, user):
+    # getting canonical domain from urls.canonical_domain(homepage) makes a HTTP request, NOT IDEAL !!!
+    six_months_ago = timezone.now() - dt.timedelta(days=SOURCE_UPDATE_DAYS_BACK)
+    sources = Source.objects.filter(
+        (Q(primary_language__isnull=True) | Q(first_story__isnull=True)),
+        name__isnull=False,
+        modified_at__lt=six_months_ago
+    ).order_by("modified_at")[:batch_size]
+
+    if not sources:
+        logger.info("No new sources to process.")
+        return updated_sources
+
+    for source in sources.iterator():
+        time.sleep(sleep_interval)  # Sleep for 0.6 seconds between requests
+        try:
+            query_str = f"canonical_domain:{source.name}"
+            provider = get_task_provider(provider_name=provider_name, api_key=None, base_url=base_url, task_name=task_name)
+
+            if task_name == "update_source_language":
+                language_samples = provider.random_sample(query_str, start_date, END_DATE, page_size=1000, fields=['media_url', 'publish_date', 'language'])
+                language_counts = Counter(record['language'] for sample in language_samples for record in sample)
+                if not language_counts:
+                    logger.warning("No languages found for source %s to analyze." % (source.name))
+                    continue
+                primary_language = language_counts.most_common(1)[0][0]
+                source.primary_language = primary_language
+                logger.info("Analyzed source %s. Primary language: %s" % (source.name, primary_language))
+
+            elif task_name == "update_publication_date":
+                results, _ = provider.paged_items(query_str, start_date, END_DATE, sort_field="publication_date", sort_order="asc", page_size=1)
+                if not results:
+                    logger.warning("No publication dates found for source %s to analyze %s." % (source.name, task_name))
+                    continue
+                first_story = dt.datetime.combine(results[0]["publish_date"], dt.datetime.min.time())
+                if first_story:
+                    first_story = timezone.make_aware(first_story)
+                    source.first_story = first_story
+                    logger.info("Analyzed source %s. First story publication date: %s" % (source.name, first_story))
+
+            source.modified_at = timezone.now()
+            updated_sources.append(source)
+        except Exception as e:
+            logger.error("Failed to analyze source %s: %s" % (source.name, str(e)))
+
+    if updated_sources:
+        field_name = 'primary_language' if task_name == "update_source_language" else 'first_story'
+        Source.objects.bulk_update(updated_sources, [field_name, "modified_at"])
+        logger.info("Completed %s extraction. Updated %d sources." % (task_name, len(updated_sources)))
+
+    return updated_sources
+
+
+@background(queue=SYSTEM_SLOW)
+def update_source_language(batch_size: int = 100) -> None:
```

Comment:
update_source_language is called in sources-update-language.py with a 'provider_name' and 'base_url' argument that aren't in this definition, so calling the command fails.